### PR TITLE
Add new Buf logo to site

### PR DESCRIPTION
**Pull request by smaye81**
This replaces the previous Buf logo in the upper right corner with the newly-designed logo.

image

### CODE REVIEW

**Comment by nicksnyder** [comment]
The vertical alignment feels off (baseline of text and logo doesn't match)

existing logo:
Screenshot 2024-07-23 at 1 51 13 PM

**Comment by smaye81** [comment]
@nicksnyder lmk what you think now.

File: src/theme/components/created-by.module.css [hunk 1]
```diff
@@ -5,6 +5,10 @@
 }
 
 .bufLogo {
+  display: flex;
+  margin-bottom: 3px;
```
Comment (nicksnyder) [inline src/theme/components/created-by.module.css:9]: I played around with this in a browser and 1px seems to align the baseline text in the logo with the text outside the logo
```suggestion
  margin-bottom: 1px;
```

Comment (smaye81) [reply src/theme/components/created-by.module.css]: It still kinda looks slightly off to me, but 🤷‍♂️

**Review by nicksnyder — APPROVED** [review]

**Pull request merged**